content/docs/console/continuous-delivery/jenkins.md

```diff
@@ -0,0 +1,145 @@
+and choosing one of the many templates that are available.
+- A bare repo and set the remote URL to be your GitHub project.
+
+## Sample Project
+
+An example project is located [here](https://github.com/pulumi/examples/azure-ts-appservice-springboot). You may download the project and upload it to your own repo to avoid having to clone the entire Pulumi Examples repo into your Jenkins workspace.
+
+## Stack and Branch Mappings
+The scripts below act on a hypothetical stack: `homer/acme/product-catalog-service-stack`.
+You can create a new stack by running `pulumi stack init` if you have already created a project.
+The source code for the stack is in a repository in GitHub and uses `TypeScript` as the language.
+
+**Note**: The names used above are purely for demonstration purposes only.
+You may choose a naming convention that best suits your organization.
+
+Alternatively, you can also run `pulumi new [template]` to create a template project.
+Learn more [here]({{< ref "/docs/reference/cli/pulumi_new.md" >}}).
+
+## `PULUMI_ACCESS_TOKEN`
+
+To login non-interactively in to the CLI, you will need to set the env var `PULUMI_ACCESS_TOKEN` as a build parameter when setting up the Jenkins build. To create a new access token, go the [Access Tokens](https://app.pulumi.com/account/tokens) page on the Pulumi Console.
+
+## Creating a New Jenkins Build
+
+### Classic UI
+
+The classic UI lets you create manual (UI) builds, as well as a Declarative pipeline-based build. There are several options available to cater to your specific setup, but for the purposes of this walkthrough, we will assume a single branch, declarative pipeline.
+
+### Jenkins Blue Ocean
+
+Blue Ocean is an overhaul of the classic Jenkins UI. It features a new and intuitive pipeline configuration UI that helps you setup a CI pipeline in a matter of just minutes. [Learn more](https://jenkins.io/projects/blueocean/) about the Jenkins Blue Ocean project.
+
+**Note** Regardless of the type of interface you choose to work with, the `Jenkinsfile` (more on that later) can be used interchangeably with both. The underlying pipeline configuration system is the same.
+
+### Plugins
+
+Ensure you have the following plugins installed:
+
+- Git
+- NodeJS
+
+You can find available plugins by navigating to the Jenkins administration page and then selecting the **Manage Plugins** option on the Manage Jenkins page.
+
+### Project Parameters (Environment Variables)
+
+In order to deploy to one of the cloud providers, you will need to ensure that the authentication environment variables are set, so that the Pulumi CLI can use them to deploy your infrastructure resources. The set of environment variables to configure vary for each cloud. For Azure, depending on your setup, you may have to set at most 4 environment variables. In this example, we will assume you are using a [Service Principal](https://www.pulumi.com/docs/reference/clouds/azure/setup/#creating-a-service-principal).
```

Comment:
```suggestion
In order to deploy to one of the cloud providers, you will need to ensure that the authentication environment variables are set, so that the Pulumi CLI can use them to deploy your infrastructure resources. The set of environment variables to configure vary for each cloud. For Azure, depending on your setup, you may have to set at most 4 environment variables. In this example, we will assume you are using a [Service Principal]({{< relref "/docs/reference/clouds/azure/setup#creating-a-service-principal" >}}).
```